api/client/samples/search.py

```diff
@@ -0,0 +1,45 @@
+"""Search example
+
+Use case: "What data does Gro have available for x?"
+
+This example shows how to query the API for sources matching a search term and
+discover what data series are available for a selected source.
+
+Note that the search functions work with any of the following entity types:
+ - metrics
+ - items
+ - regions
+ - sources (shown here)
+
+See Also
+--------
+api.client.lib.search()
+api.client.lib.search_and_lookup()
+api.client.lib.get_data_series()
+https://github.com/gro-intelligence/api-client/wiki/Entities-Definition
+https://github.com/gro-intelligence/api-client/wiki/Data-Series-Definition
+
+"""
+
+import os
+from api.client.gro_client import GroClient
+
+API_HOST = 'api.gro-intelligence.com'
+ACCESS_TOKEN = os.environ['GROAPI_TOKEN']
+
+def main():
+    client = GroClient(API_HOST, ACCESS_TOKEN)
+
+    # See what sources match the "Gro" search term
```

Comment:
Using Gro sources is a bit confusing for the first time user.   How about we  use  a more straightforward external source example which has a decent number of data series (but not huge) and is  exotic enough to be interesting, say CASDE or CONAB?

Reply:
Let's also have an example of searching for the main entities, like items or metrics.  I'm afraid having it named search.py will make people think that source-centric exploration is the right way to approach gro, which is not what we want.

Reply:
I have updated this now to showcase:

1. `search()`
2. `search_and_lookup()`
3. `get_data_series()` with item/region
4. `get_data_series()` with source